resizer.go

```diff
@@ -412,27 +414,32 @@ func shrinkImage(image *C.VipsImage, o Options, residual float64, shrink int) (*
 	return image, residual, nil
 }
 
-func shrinkJpegImage(buf []byte, input *C.VipsImage, factor float64, shrink int) (*C.VipsImage, float64, error) {
+func shrinkOnLoad(buf []byte, input *C.VipsImage, imageType ImageType, factor float64, shrink int) (*C.VipsImage, float64, error) {
 	var image *C.VipsImage
 	var err error
-	shrinkOnLoad := 1
-
-	// Recalculate integral shrink and double residual
-	switch {
-	case shrink >= 8:
-		factor = factor / 8
-		shrinkOnLoad = 8
-	case shrink >= 4:
-		factor = factor / 4
-		shrinkOnLoad = 4
-	case shrink >= 2:
-		factor = factor / 2
-		shrinkOnLoad = 2
-	}
+	shrinkFactor := 1
 
 	// Reload input using shrink-on-load
-	if shrinkOnLoad > 1 {
-		image, err = vipsShrinkJpeg(buf, input, shrinkOnLoad)
+	if shrink >= 2 {
+		// Recalculate integral shrink and double residual
+		exp := math.Floor(math.Log(float64(shrink)) / math.Log(float64(2)))
+		s := math.Pow(2, exp)
+
+		// libjpeg supports 2, 4 and 8 only.
```

Comment:
Not relevant since it's just a comment detail, but it supports `1` as well. 
Source of truth: http://www.vips.ecs.soton.ac.uk/supported/8.0/doc/html/libvips/VipsForeignSave.html#vips-jpegload